src/Server/TCPHandler.cpp

```diff
@@ -1671,11 +1673,15 @@ void TCPHandler::receiveHello()
 
         auto cred = SshCredentials(user, signature, prepare_string_for_ssh_validation(user, challenge));
         session->authenticate(cred, getClientAddress(client_info));
+        if (!default_database.empty())
+            DatabaseCatalog::instance().assertDatabaseExists(default_database);
         return;
     }
 #endif
 
     session->authenticate(user, password, getClientAddress(client_info));
+    if (!default_database.empty())
```

Comment:
I think it is better to add this check in a single place, for example after `receiveHello();` in `TCPHandler::runImpl()`